### PR TITLE
🔥 FIX - fix unit tests

### PR DESCRIPTION
**Pull request by pictos**
Match versions of `Microsoft.Maui.Controls` and `Microsoft.Maui.Controls.Compatibility`

Fixes https://github.com/CommunityToolkit/Maui/issues/1693

### CODE REVIEW

**Comment by brminnick** [comment]
@JoonghyunCho Could you help us bump Tizen to support the latest version of .NET / MAUI?

**Comment by pictos** [comment]
@JoonghyunCho from our logs

> Id: Samsung.NET.Sdk.Tizen.Manifest-8.0.200
An exception was caught: Response status code does not indicate success: 404 (The specified blob does not exist.).


Hope that helps to track the error

**Comment by JoonghyunCho** [comment]
Didn't expect to use .NET 8.0.200 for MAUI `8.0.6`. Installing Tizen Workload step is passed properly.

**Review by brminnick — DISMISSED** [review]

**Review by brminnick — DISMISSED** [review]

**Review by brminnick — DISMISSED** [review]

**Comment by brminnick** [comment]
The problem is not in our code; it builds fine locally on my machine. 

The problem is that the pipeline is installing an old version of .NET MAUI. 
Screenshot 2024-02-15 at 12 38 25 AM

I cannot think of a reason why this is happening, other than there's a problem with the workload manifests being advertised for .NET 8.0.200. 

That being said, on my local machine, when I run `dotnet workload install maui`, the correct latest version of .NET MAUI is installed. I'm unable to reproduce the problem in the build pipeline locally.

**Comment by StevilKnevil** [comment]
I've checked out and built this locally. I was able to reproduce the issue. I modified `Directory.Build.props` to have `<MauiPackageVersion>8.0.6</MauiPackageVersion>` and that seemed to resolve the issue for me

**Comment by VladislavAntonyuk** [comment]
Could you please also update System.Speech nuget package to the stable version in CommunityToolkit.Maui.Core

File: azure-pipelines.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ variables:
   PathToCommunityToolkitAnalyzersCodeFixCsproj: 'src/CommunityToolkit.Maui.Analyzers.CodeFixes/CommunityToolkit.Maui.Analyzers.CodeFixes.csproj'
   PathToCommunityToolkitMediaElementAnalyzersCodeFixCsproj: 'src/CommunityToolkit.Maui.MediaElement.Analyzers.CodeFixes/CommunityToolkit.Maui.MediaElement.Analyzers.CodeFixes.csproj'
   PathToCommunityToolkitAnalyzersUnitTestCsproj: 'src/CommunityToolkit.Maui.Analyzers.UnitTests/CommunityToolkit.Maui.Analyzers.UnitTests.csproj'
+  DotNetMauiRollbackFile: 'https://maui.blob.core.windows.net/metadata/rollbacks/8.0.6.json'
```
Comment (brminnick) [inline azure-pipelines.yml:24]: I don't love this workaround, but it's the only way I can for the hosted agent to install the latest version of .NET MAUI.

We'll hopefully be able to remove using the rollback file in a future release (or whenever this bug is fixed)

**Review by VladislavAntonyuk — APPROVED** [review]

**Comment by brminnick** [comment]
> The problem is not in our code; it builds fine locally on my machine.
> 
> The problem is that the pipeline is installing an old version of .NET MAUI:
> 
> `Manifest Version: 8.0.0-rc.1.9171/8.0.100-rc.1`
> 
> Screenshot 2024-02-15 at 12 38 25 AM
> I cannot think of a reason why this is happening, other than there's a problem with the workload manifests being advertised for .NET 8.0.200.
> 
> That being said, on my local machine, when I run `dotnet workload install maui`, the correct latest version of .NET MAUI is installed. I'm unable to reproduce the problem in the build pipeline locally.

Ok -I figured out a work-around: We can first install maui using the roll-back file; this forces the workload manifest to update. Then, when we call `dotnet workload install maui`, it installs the latest version of .NET MAUI. 

```bash
dotnet workload install maui --from-rollback-file $(DotNetMauiRollbackFile) --source https://api.nuget.org/v3/index.json
dotnet workload install maui
```

**Pull request merged**